### PR TITLE
[Automate-3537] Sequence addPolicyMembers from createToken

### PR DESCRIPTION
**Pull request by msorens**
### :nut_and_bolt: Description: What code changed, and why?

The initial pass introducing adding a token to policies during token creation made the two actions concurrent: creating the token and adding it to policies. That strategy only works well, though, if the token creation succeeds. If it fails due to an id conflict--a token of the same name already exists--then it is inappropriate for the policy assignment to still be done.

This PR sequences those so that the policy assignment is initiated iff the token creation is successful.

Let us also consider the flip side--what if token creation succeeds but the policy assignment fails? Should the two operations be done transactionally so that both must succeed otherwise neither does? I would suggest no, it is OK to create the token regardless, and if the policy assignments failed they could be retried through other routes. (Partly because that seems reasonable to me and partly because the implementation of this is straightforward; hence, this PR. 😁 )

### :chains: Related Resources
PR #3511 (Introduce policy dropdown for tokens)

### :+1: Definition of Done
Policy assignments are not attempted if token creation fails for any reason.

### :athletic_shoe: How to Build and Test the Change
1. Rebuild automate-ui.
2. Navigate to Settings >> Tokens
3. Use <kbd>Create Token</kbd> to create an initial token, t1; neither projects nor policies matters.
4. Again open up the token creation modal with Use <kbd>Create Token</kbd>. Enter the same id for the token and this time be sure to select one or more policies from the dropdown.
5. Upon pressing <kbd>Create</kbd> you should see an in-modal error notification and on the underlying page **_no_** notification for policies.
6. Change the id to something unique so the token creation will succeed. This time observe that the modal closes, the first notification indicates token created, and then there is an additional success notification for each policy you specified.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts [hunk 1]
```diff
@@ -87,6 +87,7 @@ export class ApiTokenListComponent implements OnInit, OnDestroy {
       .subscribe(state => {
         this.creatingToken = false;
         if (state === EntityStatus.loadingSuccess) {
+          this.assignPolicies();
```
Comment (msorens) [inline components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts:90]: Previously, this call (inline) to dispatch AddPolicyMembers was done at the same time the CreateToken action was dispatched. By moving it here, we now wait for a successful notification of token creation before starting the policy additions.

File: components/automate-ui/src/app/modules/token/token-list/api-token-list.component.spec.ts [hunk 2]
```diff
@@ -90,18 +93,48 @@ describe('ApiTokenListComponent', () => {
...
+      component.createModalVisible = true;
       spyOn(store, 'dispatch').and.callThrough();
       component.createTokenForm.controls.policies.setValue(null);
       component.createToken();
       expect(store.dispatch).toHaveBeenCalledTimes(1);
+
+      store.dispatch(new CreateTokenSuccess({ 'id': 't1', 'projects': [] } as ApiToken));
+      fixture.detectChanges();
+
+      const setupCallsToDispatch = 2;
```
Comment (msorens) [inline components/automate-ui/src/app/modules/token/token-list/api-token-list.component.spec.ts:105]: That's one from calling `createToken` plus one explicitly called on L102 above.

File: components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts [hunk 3]
```diff
@@ -132,22 +133,20 @@ export class ApiTokenListComponent implements OnInit, OnDestroy {
...
-        this.store.dispatch(new AddPolicyMembers(<PolicyMembersMgmtPayload>{
-          id,
-          members: [member]
-        }))
-      );
-    }
+  private assignPolicies(): void {
+    const member = stringToMember(`token:${this.createTokenForm.controls.id.value}`);
+    const policies: string[] = this.createTokenForm.controls.policies.value || [];
+    policies.forEach(id => this.store.dispatch(new AddPolicyMembers(<PolicyMembersMgmtPayload>{
```
Comment (srenatus) [inline components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts:146]: ❔ What does the cast give us here? (Sorry, I can see it's moved code, just wondering.)

File: components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts [hunk 4]
```diff
@@ -132,22 +133,20 @@ export class ApiTokenListComponent implements OnInit, OnDestroy {
...
-    if (policies) {
-      policies.forEach(id =>
-        this.store.dispatch(new AddPolicyMembers(<PolicyMembersMgmtPayload>{
-          id,
-          members: [member]
-        }))
-      );
-    }
+  private assignPolicies(): void {
+    const member = stringToMember(`token:${this.createTokenForm.controls.id.value}`);
```
Comment (srenatus) [inline components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts:144]: Would this look cleaner if the id was passed? `assignPolicies(tokenID: string, policyIDs: string[]): void`? What do you think?

File: components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts (hunk 3 above)
Comment (msorens) [reply components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts:146]: Actually, it is unneeded here--good eye!

File: components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts (hunk 4 above)
Comment (msorens) [reply components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts:144]: I agree with you: on one hand, it shows a bit more... intent. But on the other hand (which is why I chose to do it this way), I prefer my subscriptions (the call site for this) to be more "high-level" in a sense, typically just dispatching something or setting a property to open/close something, then let the helper functions have the nitty-gritty details.

**Pull request merged**